controllers/application/ingress_gateway.go

```diff
@@ -109,13 +109,9 @@ func (r *ApplicationReconciler) reconcileIngressGateway(ctx context.Context, app
 	return reconcile.Result{}, err
 }
 
-// Filter for gateways named like *-ingress
+// Filter for gateways named like *-ingress-*
 func isIngressGateway(gateway *networkingv1beta1.Gateway) bool {
-	segments := strings.Split(gateway.Name, "-")
+	match, _ := regexp.MatchString("^.*-ingress-.*$", gateway.Name)
```

Comment:
Is it possible to write a test for verifying that it no longer creates and re-creates gateways like a madman?

Reply:
Im looking into some more advanced tests in #118.

This PR really only allows applications to have dashes in their names, and does not change the creation/re-creation of madman-gateways. Should be fixed in #118 though :)